### PR TITLE
feat: create notification for user top reader badges

### PR DESCRIPTION
**Pull request by omBratteng**
AS-667

### CODE REVIEW

File: src/notifications/builder.ts [hunk 1]
```diff
@@ -267,6 +278,19 @@ export class NotificationBuilder {
     return this;
   }
 
+  avatarTopReaderBadge(
+    ctx: NotificationUserTopReaderContext,
+  ): NotificationBuilder {
+    this.avatars.push({
+      type: 'top_reader_badge',
+      name: ctx.keyword.flags.title,
+      targetUrl: '',
```
Comment (rebelchris) [inline src/notifications/builder.ts:287]: This needs something like notification?topreader&badgeId no?

File: src/notifications/generate.ts [hunk 2]
```diff
@@ -127,12 +128,27 @@ export const notificationTitleMap: Record<
...
 export const generateNotificationMap: Record<
   NotificationType,
   (builder: NotificationBuilder, ctx: never) => NotificationBuilder
 > = {
+  user_given_top_reader: (builder, ctx: NotificationUserTopReaderContext) =>
+    builder
+      .icon(NotificationIcon.TopReaderBadge)
+      .referenceUserTopReader(ctx.userTopReader)
+      .targetUrl(notificationsLink)
+      .setTargetUrlParameter([
```
Comment (rebelchris) [inline src/notifications/generate.ts:146]: Oh I see you do it here, maybe we should do this in the builder rather?

**Review by rebelchris — CHANGES_REQUESTED** [review]
Minor comments

File: src/notifications/builder.ts (hunk 1 above)
Comment (omBratteng) [reply src/notifications/builder.ts:287]: I don't use that URL in frontend, but it has to be set in DB, se empty string it is 😬

File: src/notifications/generate.ts (hunk 2 above)
Comment (omBratteng) [reply src/notifications/generate.ts:146]: Not sure what you mean? Doing it like this for `streak_reset_restore`, `promoted_to_admin` and `promoted_to_moderator` notifications.

Comment (rebelchris) [reply src/notifications/generate.ts:146]: Oh right, I thought we set it in the builder already, that case never mind me :)

**Review by rebelchris — APPROVED** [review]

**Comment by omBratteng** [comment]
Forgot to add email notification 🙈

**Comment by omBratteng** [comment]
Added test

**Review by sshanzel — APPROVED** [review]
Looks good to me! :shipit: 

Just a question: where do we trigger the generation of this notification?

**Comment by omBratteng** [comment]
> Just a question: where do we trigger the generation of this notification?

@sshanzel black magic? 😈

It's going to be triggered via a cronjob I'm working on

**Pull request merged**